### DISCUSSION TITLE
Liveness checking for DB

### DISCUSSION THREAD

**Discussion by susanta-mukhopadhyay** [9tigerio/db2rest#837] [General]
How are we checking liveness of database? Do we have any implementation for that?

**Comment by thadguidry** [comment]
Most monitoring solutions (Sensu, Nagios, etc.) do a DB liveness check by simply attempting a login connect to the DB with blunt "user=user ; password=pass".
Example: https://github.com/sensu-plugins/sensu-plugins-postgres/blob/master/bin/check-postgres-alive.rb

But liveness check is not the only thing often needed in monitoring a DB (locks, stats, replication, dbsize, connections, are often needed to be quickly checked or looked at when slow queries or long timeouts start occurring).

But DB monitoring itself is not something DB2Rest concerns itself about or should.  That's for other open-source monitoring solutions to provide and already do quite well.

**Reply by susanta-mukhopadhyay** [reply]
Yes, true, DB monitoring may be separate concern. I was thinking of we only have some mechanism to check whether application would not hang-up due to unavailability of DB.

**Comment by kdhrubo** [comment]
DB2REST actuator should show the metrics related to database. Hikari may
provide additional checks as part of configuration

**Reply by susanta-mukhopadhyay** [reply]
Yes, I think Hikari supports timeout to check DB. If it is configured, it will prevent hang-up of the application.

**Comment by kdhrubo** [comment]
It will error out. As pool will complain

**Comment by kdhrubo** [comment]
Closing this as actuator is turned on.
